cfgov/v1/util/migrations.py

```diff
@@ -0,0 +1,59 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+from treebeard.mp_tree import MP_Node
+
+
+def get_page(apps, slug):
+    base_page_cls = apps.get_model('wagtailcore', 'Page')
+    return base_page_cls.objects.get(slug=slug)
+
+
+def get_free_path(apps, parent_page):
+    offset = 1
+    base_page_cls = apps.get_model('wagtailcore', 'Page')
+
+    while True:
```

Comment:
@vccabral thanks for the feedback.

I'm not sure how to make that while loop iterative, considering it has to increase until it finds a free number. You could do something like this:

```py
import itertools

for i in itertools.count():
    # do stuff and return if i is free
```

but my sense is that this is less clear, especially to people who aren't familiar with the somewhat obscure [`itertools.count`](https://docs.python.org/2/library/itertools.html#itertools.count).

I don't believe there are mixed spaces and tabs - what makes you think so?

The use of this:

```py
try:
    Model.objects.get(foo=bar)
except Model.DoesNotExist:
    # do something
```

instead of

```py
if not Model.objects.filter(foo=bar).exists():
```

seems somewhat of a style choice to me. I think the first way reads a little clearer. I've also seen the Django codebase use the `try/except` style, for example [here](https://github.com/django/django/blob/1.8.13/django/views/generic/detail.py#L51) or [here](https://github.com/django/django/blob/4f336f66523001b009ab038b10848508fd208b3b/tests/select_related/tests.py#L25).

Regarding pulling out to a common lib, that's why I put these methods in `v1.util.migrations` instead of the migration file itself.